packages/toolkit/src/entities/unsorted_state_adapter.ts

```diff
@@ -158,7 +158,9 @@ export function createUnsortedStateAdapter<T>(
         0
 
       if (didMutateIds) {
-        state.ids = state.ids.map((id) => newKeys[id] || id)
+        state.ids = Array.from(
```

Comment:
I works well. Here's my suggestion:
```suggestion
        state.ids = Object.keys(state.entities)
```